--- origsrc/tigervnc-1.0.0/java/src/com/tigervnc/vncviewer/MANIFEST.MF	2009-08-20 04:46:42.000000000 -0500
+++ src/tigervnc-1.0.0/java/src/com/tigervnc/vncviewer/MANIFEST.MF	2009-11-15 16:16:53.511443400 -0600
@@ -1,2 +1,2 @@
 Manifest-Version: 1.0
-Main-Class: VncViewer
+Main-Class: com.tigervnc.vncviewer.VncViewer
--- origsrc/tigervnc-1.0.0/java/src/com/tigervnc/vncviewer/index.vnc	2009-08-20 04:46:42.000000000 -0500
+++ src/tigervnc-1.0.0/java/src/com/tigervnc/vncviewer/index.vnc	2009-11-15 16:29:03.365916500 -0600
@@ -15,7 +15,7 @@
 <TITLE>
 $USER's $DESKTOP desktop ($DISPLAY)
 </TITLE>
-<APPLET CODE=VncViewer.class ARCHIVE=VncViewer.jar
+<APPLET CODE=com/tigervnc/vncviewer/VncViewer.class ARCHIVE=VncViewer.jar
         WIDTH=$APPLETWIDTH HEIGHT=$APPLETHEIGHT>
 <param name=PORT value=$PORT>
 $PARAMS
