--- origsrc/tigervnc-1.2.0/cmake/BuildPackages.cmake	2011-11-08 03:16:36.000000000 -0600
+++ src/tigervnc-1.2.0/cmake/BuildPackages.cmake	2012-09-03 04:25:55.979417000 -0500
@@ -86,5 +86,3 @@ endif() #UNIX
 # Common
 #
 
-install(FILES ${CMAKE_SOURCE_DIR}/LICENCE.TXT DESTINATION doc)
-install(FILES ${CMAKE_SOURCE_DIR}/README.txt DESTINATION doc)
--- origsrc/tigervnc-1.2.0/java/CMakeLists.txt	2012-03-09 14:34:29.000000000 -0600
+++ src/tigervnc-1.2.0/java/CMakeLists.txt	2012-09-03 04:20:16.390822200 -0500
@@ -97,5 +97,5 @@ if(NOT WIN32)
     ${JAVA_SRCDIR}/README
     ${JAVA_SRCDIR}/index.vnc
     ${JAVA_SRCDIR}/favicon.ico
-    DESTINATION vnc/classes)
+    DESTINATION share/vnc/classes)
 endif()
