--- origsrc/tigervnc-1.2.0/common/fltk/src/CMakeLists.txt	2012-01-17 16:33:45.000000000 -0600
+++ src/tigervnc-1.2.0/common/fltk/src/CMakeLists.txt	2012-09-03 02:04:30.722637400 -0500
@@ -191,7 +191,7 @@ if(HAVE_XINERAMA)
 endif(HAVE_XINERAMA)
 
 if(USE_XFT)
-   target_link_libraries(fltk_static ${X11_Xft_LIB})
+   target_link_libraries(fltk_static ${X11_Xft_LIB} fontconfig)
 endif(USE_XFT)
 
 if(HAVE_XFIXES)
