--- origsrc/tigervnc/unix/xserver/hw/vnc/Makefile.am	2010-09-16 21:39:32.644003800 -0500
+++ src/tigervnc/unix/xserver/hw/vnc/Makefile.am	2010-09-16 23:50:52.841579900 -0500
@@ -17,7 +17,7 @@ libvnccommon_la_SOURCES = $(HDRS) vncExt
 
 libvnccommon_la_CPPFLAGS = -DVENDOR_RELEASE="$(VENDOR_RELEASE)" \
 	-DVENDOR_STRING="\"$(VENDOR_STRING)\"" -I$(TIGERVNC_SRCDIR)/common -UHAVE_CONFIG_H \
-	-I$(TIGERVNC_SRCDIR)/unix/vncconfig $(XVNC_CPPFLAGS) -I$(includedir)/pixman-1 -I$(includedir)
+	-I$(TIGERVNC_SRCDIR)/unix/vncconfig $(XVNC_CPPFLAGS) -I$(top_srcdir)/include -I$(includedir)/pixman-1 -I$(includedir)
 
 bin_PROGRAMS = Xvnc
 
@@ -37,6 +37,8 @@ Xvnc_LDADD = $(XVNC_LIBS) libvnccommon.l
 
 Xvnc_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
 
+if XORG
+
 libvnc_la_LTLIBRARIES = libvnc.la
 libvnc_ladir = $(moduledir)/extensions
 
@@ -52,6 +54,8 @@ libvnc_la_LDFLAGS = -module -avoid-versi
 
 libvnc_la_LIBADD = libvnccommon.la $(COMMON_LIBS)
 
+endif XORG
+
 EXTRA_DIST = Xvnc.man
 
 # C++ hacks
