--- xorg-server-1.5.2.orig/unix/xserver/Makefile.am	2008-10-10 13:12:48.000000000 -0500
+++ xorg-server-1.5.2/unix/xserver/Makefile.am	2008-10-29 17:17:56.692163800 -0500
@@ -62,11 +62,13 @@
 	config \
 	hw
 
+if XORG
 aclocaldir = $(datadir)/aclocal
 aclocal_DATA = xorg-server.m4
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = xorg-server.pc
+endif
 
 EXTRA_DIST = xorg-server.pc.in xorg-server.m4 ChangeLog autogen.sh
 
--- xorg-server-1.6.0.orig/unix/xserver/acinclude.m4	2008-12-16 14:03:03.000000000 -0600
+++ xorg-server-1.6.0/unix/xserver/acinclude.m4	2009-02-25 23:06:00.763625000 -0600
@@ -29,6 +29,9 @@ i?86-*-linux*|x86_64-*-linux*|powerpc-*-
 i?86-apple-darwin*)
     pic_options='-fno-common'
     ;;
+*-*-cygwin*|*-*-mingw32)
+    pic_options='-DDLL_EXPORT'
+    ;;
 *)
     dolt_supported=no
     ;;
--- xorg-server-1.5.3.orig/unix/xserver/configure.ac	2009-01-18 18:30:38.863875000 -0600
+++ xorg-server-1.5.3/unix/xserver/configure.ac	2009-01-18 18:33:51.692000000 -0600
@@ -467,7 +467,7 @@ AC_ARG_WITH(builderstring,   AS_HELP_STR
 AC_ARG_WITH(fontdir,         AS_HELP_STRING([--with-fontdir=FONTDIR], [Path to top level dir where fonts are installed (default: ${libdir}/X11/fonts)]),
                                 [ FONTDIR="$withval" ],
                                 [ FONTDIR="${libdir}/X11/fonts" ])
-DEFAULT_FONT_PATH="${FONTDIR}/misc/,${FONTDIR}/TTF/,${FONTDIR}/OTF,${FONTDIR}/Type1/,${FONTDIR}/100dpi/,${FONTDIR}/75dpi/"
+DEFAULT_FONT_PATH="${FONTDIR}/misc/,${FONTDIR}/TTF/,${FONTDIR}/OTF,${FONTDIR}/Type1/,${FONTDIR}/100dpi/,${FONTDIR}/75dpi/,built-ins"
 case $host_os in
 	darwin*)    DEFAULT_FONT_PATH="${DEFAULT_FONT_PATH},/Library/Fonts,/System/Library/Fonts" ;;
 esac
